src/Fetch.js

```diff
@@ -567,21 +585,27 @@ function startFetch(fetch, successcb, errorcb, progresscb, readystatechangecb) {
 }
 
 function fetchGetResponseHeadersLength(id) {
-    return lengthBytesUTF8(Fetch.xhrs[id-1].getAllResponseHeaders()) + 1;
+    return lengthBytesUTF8(Fetch.xhrs[id].getAllResponseHeaders()) + 1;
 }
 
 function fetchGetResponseHeaders(id, dst, dstSizeBytes) {
-    var responseHeaders = Fetch.xhrs[id-1].getAllResponseHeaders();
+    var responseHeaders = Fetch.xhrs[id].getAllResponseHeaders();
     var lengthBytes = lengthBytesUTF8(responseHeaders) + 1;
     stringToUTF8(responseHeaders, dst, dstSizeBytes);
     return Math.min(lengthBytes, dstSizeBytes);
 }
 
 //Delete the xhr JS object, allowing it to be garbage collected.
 function fetchFree(id) {
-  //Note: should just be [id], but indexes off by 1 (see: #8803)
 #if FETCH_DEBUG
-  console.log("fetch: Deleting id:" + (id-1) + " of " + Fetch.xhrs);
+    console.log("fetch: Deleting id:" + id + " of " + Fetch.xhrs);
 #endif
-  delete Fetch.xhrs[id-1];
+    var xhr = Fetch.xhrs[id];
+    if (xhr !== undefined) {
```

Comment:
Maybe just `if (xhr)` for codesize?  (since we know that no other falsely things could be in the map).